src/actions/index.js

```diff
@@ -371,6 +372,44 @@ function updateOutput(
     const {config, layout, graphs, dependenciesRequest, hooks} = getState();
     const {InputGraph} = graphs;
 
+    const getThisRequestIndex = () => {
```

Comment:
Comment not specific to this review but `updateOutput` is in need some deep cleanup. If `TreeContainer` is the rendere's heart, this is certainly the brain. Opening https://github.com/plotly/dash-core/issues/56 for follow up.